packages/ckeditor5-engine/src/view/domconverter.ts

```diff
@@ -1787,6 +1792,24 @@ function _logUnsafeElement( elementName: string ): void {
 	}
 }
 
+// In certain cases, Firefox mysteriously assigns so called "restricted objects" to native DOM Range properties.
+// Any attempt at accessing restricted object's properties causes errors.
+// https://github.com/ckeditor/ckeditor5/issues/9635
+function isGeckoRestrictedDomSelection( domSelection: DomSelection ): boolean {
+	if ( !env.isGecko ) {
+		return false;
+	}
+
+	try {
+		const container = domSelection.getRangeAt( 0 ).startContainer;
```

Comment:
There is a valid case when the DOM selection has no ranges and this should not affect this function. Now it will return true for selection without ranges and it would not be true.